terminfo/terminfo.go

```diff
@@ -167,59 +167,61 @@ type Terminfo struct {
-	KeyMetaRight    string // meta-right
-	KeyMetaLeft     string // meta-left
-	KeyAltUp        string // alt-up
-	KeyAltDown      string // alt-left
-	KeyAltRight     string // alt-right
-	KeyAltLeft      string // alt-left
-	KeyCtrlHome     string
-	KeyCtrlEnd      string
-	KeyMetaHome     string
-	KeyMetaEnd      string
-	KeyAltHome      string
-	KeyAltEnd       string
-	KeyAltShfUp     string
-	KeyAltShfDown   string
-	KeyAltShfLeft   string
-	KeyAltShfRight  string
-	KeyMetaShfUp    string
-	KeyMetaShfDown  string
-	KeyMetaShfLeft  string
-	KeyMetaShfRight string
-	KeyCtrlShfUp    string
-	KeyCtrlShfDown  string
-	KeyCtrlShfLeft  string
-	KeyCtrlShfRight string
-	KeyCtrlShfHome  string
-	KeyCtrlShfEnd   string
-	KeyAltShfHome   string
-	KeyAltShfEnd    string
-	KeyMetaShfHome  string
-	KeyMetaShfEnd   string
-	EnablePaste     string // bracketed paste mode
-	DisablePaste    string
-	PasteStart      string
-	PasteEnd        string
-	Modifiers       int
-	InsertChar      string // string to insert a character (ich1)
-	AutoMargin      bool   // true if writing to last cell in line advances
-	TrueColor       bool   // true if the terminal supports direct color
+	StrikeThrough    string // smxx
+	SetFgBg          string // setfgbg
+	SetFgBgRGB       string // setfgbgrgb
+	SetFgRGB         string // setfrgb
+	SetBgRGB         string // setbrgb
+	KeyShfUp         string // shift-up
+	KeyShfDown       string // shift-down
+	KeyShfPgUp       string // shift-kpp
+	KeyShfPgDn       string // shift-knp
+	KeyCtrlUp        string // ctrl-up
+	KeyCtrlDown      string // ctrl-left
+	KeyCtrlRight     string // ctrl-right
+	KeyCtrlLeft      string // ctrl-left
+	KeyMetaUp        string // meta-up
+	KeyMetaDown      string // meta-left
+	KeyMetaRight     string // meta-right
+	KeyMetaLeft      string // meta-left
+	KeyAltUp         string // alt-up
+	KeyAltDown       string // alt-left
+	KeyAltRight      string // alt-right
+	KeyAltLeft       string // alt-left
+	KeyCtrlHome      string
+	KeyCtrlEnd       string
+	KeyMetaHome      string
+	KeyMetaEnd       string
+	KeyAltHome       string
+	KeyAltEnd        string
+	KeyAltShfUp      string
+	KeyAltShfDown    string
+	KeyAltShfLeft    string
+	KeyAltShfRight   string
+	KeyMetaShfUp     string
+	KeyMetaShfDown   string
+	KeyMetaShfLeft   string
+	KeyMetaShfRight  string
+	KeyCtrlShfUp     string
+	KeyCtrlShfDown   string
+	KeyCtrlShfLeft   string
+	KeyCtrlShfRight  string
+	KeyCtrlShfHome   string
+	KeyCtrlShfEnd    string
+	KeyAltShfHome    string
+	KeyAltShfEnd     string
+	KeyMetaShfHome   string
+	KeyMetaShfEnd    string
+	EnablePaste      string // bracketed paste mode
+	DisablePaste     string
+	PasteStart       string
+	PasteEnd         string
+	Modifiers        int
+	InsertChar       string // string to insert a character (ich1)
+	AutoMargin       bool   // true if writing to last cell in line advances
+	TrueColor        bool   // true if the terminal supports direct color
+	StringTerminator string
```

Comment:
My instinct is that this should be a parameterized capability.  Much like we do for cursor location, etc.  Rather than two separate entries in the terminfo.

I need to look closely as I'm not sure we have support for string parameters.

Reply:
So terminfo supports string parameters, but my implementation assumes integers.  (See TParam).  I think we need a new API for this.

Reply:
Or possibly TParam could use type assertions, but I worry about that making it slower -- doing a type assertion each time you draw a cell on the screen would be unfortunate.